src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs

```diff
@@ -3090,20 +3087,36 @@ var projectUniqueNamesForBuildIntToUpdate
                     .Distinct()
                     .ToList();
 
-                var successfulFrameworks = allFrameworks
-                    .Except(unsuccessfulFrameworks)
-                    .ToList();
+                var originalFrameworks = updatedPackageSpec
+                    .TargetFrameworks
+                    .ToDictionary(x => x.FrameworkName, x => x.TargetAlias);
+
+                List<(NuGetProjectAction, BuildIntegratedInstallationContext)> projectActionsAndInstallationContexts = new(nuGetProjectActions.Length);
+
+                foreach (var action in nuGetProjectActions)
+                {
+                    PackageSpec referencePackageSpec = null;
+                    referencePackageSpec = action.NuGetProjectActionType switch
+                    {
+                        NuGetProjectActionType.Install or NuGetProjectActionType.Update => restoreResult.Result.LockFile.PackageSpec,
+                        NuGetProjectActionType.Uninstall => originalPackageSpec,
+                        _ => throw new InvalidOperationException("Unknown NuGetProjectActionType"),
```

Comment:
Should we localize this exception message?

Reply:
IMO, this one won't benefit. The switch statement already exhausts every defined `NuGetProjectActionType` enum value, but because it's valid, from a compiler's point of view, to do `(NuGetProjectActionType)1234`, the compiler forces you to still have a default value, just in case. But that would be a programming error, one that we wouldn't expect a customer to ever see.

Reply:
Yep, these are exceptions that basically tell us that we haven't implemented all the scenarios correctly.